File: 프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.py
def solution(my_string, overwrite_string, s):
    answer = ''
    i = 0
    
    while i < len(my_string):
        if i == s:
            answer = answer + overwrite_string
            i = i + len(overwrite_string)
        else:
            answer = answer + my_string[i]
            i = i + 1
        
    return answer

File: 프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.java
class Solution {
    public String solution(String my_string, String overwrite_string, int s) {
        String answer = "";
        char[] arr = my_string.toCharArray();
        
        for(int i=0; i<arr.length; i++){
            if(i != s){
                answer += arr[i];
            } else {
                answer += overwrite_string;
                i += overwrite_string.length() - 1;
            }
        }
        
        return answer;
    }
}